interface/src/app/sample_annotation/annotation.spec.js

```diff
@@ -0,0 +1,68 @@
+describe('sample_annotation', function() {
+  var $httpBackend, $log, SampleMocks;
+
+  beforeEach(module('adage.sampleAnnotation'));
+  beforeEach(module('adage.mocks.sample'));
+  beforeEach(inject(function(_$httpBackend_, _$log_, _SampleMocks_) {
+    $httpBackend = _$httpBackend_;
+    $log = _$log_;
+    SampleMocks = _SampleMocks_;
+  }));
+
+  // Ensure no outstanding expectation or request at the end.
+  afterEach(function() {
+    $httpBackend.verifyNoOutstandingExpectation();
+    $httpBackend.verifyNoOutstandingRequest();
+    // very useful $log capture information from:
+    // http://www.jvandemo.com/how-to-access-angular-log-debug-messages-from-within-karma/
+    if ($log.debug.logs.length > 0) {
+      console.log($log.debug.logs);
+    }
+  });
+
+  describe('SampleAnnotationCtrl', function() {
+    var SampleAnnotationCtrl, $location, $scope, mockStateParams, $log;
```

Comment:
Actually here `$location`, `$scope`, `mockStateParams` and `$log` are not necessary, unless you want to do more in the  first test case ('should pass a dummy test'). This line in the dummy test:
```
SampleAnnotationCtrl = $controller('SampleAnnotationCtrl');
```
will automatically figure out the dependencies of `SampleAnnotationCtrl` and inject them in. (Besides, `$log` has  already been defined in the parent `describe` block.) So unless you need to use these dependencies directly (like the call of $log in afterEach block above), you don't need them.

And the beforeEach() on lines 26-39 can be simplified into:
```
beforeEach(inject(function($controller) {
  SampleAnnotationCtrl = $controller('SampleAnnotationCtrl');
}));
```